### PR TITLE
Redefine `split.step` to skip samples and not unique values

### LINKED ISSUE #55: Dummy variables ignored for split.step > 1
I realized that dummy policy variables are ignored when policy_tree is specified with split.step > 1. This makes of course sense when having a closer look at the algorithm, but it is an unfortunate feature for practitioners that might not be aware of this. I solve it currently brute force by rounding the continuous variables and proceed with split.step = 1.

### PR DESCRIPTION
**Pull request by erikcs**
See issue #55:

This implements the suggested resolution: `split.step` now skips observations instead of unique values.

### CODE REVIEW

**Comment by erikcs** [comment]
The changed logic should be very simple @kanodiaayush, it's in 7a24cea

**Review by kanodiaayush — APPROVED** [review]
LGTM, approval.

Sorry this took shorter than I thought it would. Maybe in the PR if you could add LOC that would "nudge" me. 

Do you also see an issue with indenting? I find it hard to follow indents. Do you have a suggestion on how I could do that better? Perhaps we move to 4 space indent?

**Comment by erikcs** [comment]
Where's the issue with indenting? It should be 2 spaces everywhere?

**Comment by kanodiaayush** [comment]
There should not be an issue - just that its not clear enough to get which
for loop we are in. Maybe github can have an indent extension which marks
this out; or maybe we can use 4 space indent to improve this. i think
ignore this comment.
Ayush Kanodia,

PhD Student,
Stanford Computer Science
